### PR TITLE
Loggly: don't treat colon in url as seperator

### PR DESCRIPTION
**Pull request by fffw**
Fixes https://github.com/getlantern/lantern/issues/2322

### CODE REVIEW

File: src/github.com/getlantern/flashlight/logging/logging.go [hunk 1]
```diff
@@ -179,24 +179,28 @@ func (w logglyErrorWriter) Write(b []byte) (int, error) {
...
-		message = parts[1]
-	default:
-		message = parts[pl-2] + ":" + parts[pl-1]
+	lastColonPos := -1
+	colonsSeen := 0
+	for p := len(fullMessage) - 2; p >= 0; p-- {
+		if fullMessage[p] == ':' {
+			lastChar := fullMessage[p+1]
+			// to prevent colon in "http://" and "x.x.x.x:80" be treated as seperator
+			if !(lastChar == '/' || lastChar >= '0' && lastChar <= '9') {
```
Comment (oxtoacart) [inline src/github.com/getlantern/flashlight/logging/logging.go]: I think that this is too specific.  If you do what I originally suggested, which is to take as many chunks as will fit into 100 characters, then that should largely take care of this problem.

Comment (fffw) [inline src/github.com/getlantern/flashlight/logging/logging.go]: That will leave too many message groups of which the only difference is the host to connect, like below

```
fronted: HTTP ERROR FOR MASQUERADE ritegamer.com: Head http://www.google.com/humans.txt: Unable to dial proxy: Unable to dial proxy: Unable to dial masquerade ritegamer.com: tlsdialer: DialWithDialer timed out\n
```

Comment (oxtoacart) [inline src/github.com/getlantern/flashlight/logging/logging.go]: True.  At the end of the day, once we're in the position of having to parse freeform text, it's kind of a losing proposition.  There's a separate ticket somewhere to start recording the line number on which a message was logged.  What if we just use that to group things?

Comment (fffw) [inline src/github.com/getlantern/flashlight/logging/logging.go]: Hm, the motive of truncating message is not to miss some item when grouping by it (https://github.com/getlantern/lantern/issues/2287), and we always have `fullMessage`. Grouping by location and by reason can be of different dimension. Anyway, it is insignificant I believe.

Comment (oxtoacart) [inline src/github.com/getlantern/flashlight/logging/logging.go]: Okay, sold.

**Pull request merged**